### PR TITLE
[RetryLink] add 409 and 429

### PR DESCRIPTION
**Pull request by alangenfeld**
add rate limiting and idempotency conflict status codes to be retried

### CODE REVIEW

**Comment by alangenfeld** [comment]
* **#22970** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22970?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alangenfeld and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: js_modules/dagster-ui/packages/ui-core/src/app/AppProvider.tsx [hunk 1]
```diff
@@ -44,6 +44,8 @@ import './blueprint.css';
 // when you copy-paste so they don't get pasted into editors, etc.
 patchCopyToRemoveZeroWidthUnderscores();
 
+const httpStatusCodesToRetry = [502, 503, 504, 429, 409];
```
Comment (alangenfeld) [inline js_modules/dagster-ui/packages/ui-core/src/app/AppProvider.tsx:47]: one concern is that for `429` we send a `retry-after` header which we certainly don't  respect here in our code, not sure if `RetryLink` under the hood will

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/app/AppProvider.tsx:47]: It won't under the hood but we can customize RetryLink to do this.

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/app/AppProvider.tsx:47]: We could do that by changing `delay` to a function and I *think* the `error` argument should have the header accessible https://www.apollographql.com/docs/react/api/link/apollo-link-retry/#custom-strategies

**Review by gibsondan — CHANGES_REQUESTED** [review]
doing some queue clearing for older PRs that went stale - I think this ended up requiring some changes to RetryLink?

**Pull request closed without merging**